packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -252,46 +353,174 @@ export class OwnedAddressTCRWithAppeals extends BaseWrapper<OwnedAddressTCRWithA
+    return (appealExpiryDate > new Date());
+  }
+
+  /**
+   * Gets the expiry time of the appeal phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getAppealExpiryTimestamp(listingAddress: EthAddress): Promise<BigNumber> {
+    return this.instance.getAppealPhaseExpiry.callAsync(listingAddress);
+  }
+
+  /**
+   * Gets the expiry time of the commit vote phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getCommitVoteExpiryTimestamp(listingAddress: EthAddress): Promise<BigNumber> {
+    // if there is no challenge
+    const challenge = await this.instance.getListingChallengeID.callAsync(listingAddress);
+    if (challenge.toNumber() !== 0) {
+      const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
+      const revealPeriodActive = await voting.getCommitPeriodExpiry(challenge);
+
+      return revealPeriodActive;
+    } else {
+      return new BigNumber(0);
+    }
+  }
+
+  /**
+   * Gets the expiry time of the commit vote phase.
+   * @param listingAddress Address of listing to check
+   */
+  public async getRevealVoteExpiryTimestamp(listingAddress: EthAddress): Promise<BigNumber> {
+    // if there is no challenge
+    const challenge = await this.instance.getListingChallengeID.callAsync(listingAddress);
+    if (challenge.toNumber() !== 0) {
+      const voting = Voting.atUntrusted(this.web3Wrapper, await this.instance.voting.callAsync());
+      const revealPeriodActive = await voting.getRevealPeriodExpiry(challenge);
+
+      return revealPeriodActive;
+    } else {
+      return new BigNumber(0);
+    }
+  }
+
   /**
    * Get address for token used with this TCR
    */
   public async getTokenAddress(): Promise<EthAddress> {
     return this.instance.token.callAsync();
   }
 
+  /**
+   * Get address for voting contract used with this TCR
+   */
+  public async getVotingAddress(): Promise<EthAddress> {
+    return this.instance.voting.callAsync();
+  }
+
   /**
    * Checks if an address has application associated with it
    * App can either be in progress or approved
-   * @param address Address of potential listing to check
+   * @param address Address of listing to check
    */
   public async appWasMade(listingAddress: EthAddress): Promise<boolean> {
     return this.instance.appWasMade.callAsync(listingAddress);
   }
 
   /**
    * Checks if an address has application/listing has an unresolved challenge
-   * @param address Address of potential listing to check
+   * @param address Address of listing to check
    */
   public async challengeExists(listingAddress: EthAddress): Promise<boolean> {
     return this.instance.challengeExists.callAsync(listingAddress);
   }
 
+  /**
+   * Gets the current challenge ID of the listing. Returns 0 if no current challenge.
+   * @param listingAddress Address of listing to check
+   */
+  public async getListingChallengeID(listingAddress: EthAddress): Promise<BigNumber> {
```

Comment:
how about create a Challenge object/class and just operate on that, instead of having state managed in the TCR class

Reply:
not really sure this is necessary right now. Would you think we should separate `Listing` into its own class as well?

Reply:
Add a clubhouse technical-debt card and let's merge this one as is. It would get too big